src/core/compact.rs

```diff
@@ -85,153 +89,172 @@ pub trait FindexCompact<
-        // database.
-        let mut chain_table_adds = EncryptedTable::default();
-        let mut entry_table = EntryTable::decrypt::<BLOCK_LENGTH, DEM_KEY_LENGTH, DemScheme>(
-            &k_value,
-            &encrypted_entry_table,
-        )?;
-        // Entry Table items indexing empty chains should be removed.
-        let mut entry_table_uids_to_drop = Vec::new();
+        let all_uids_vec = Vec::from_iter(all_uids.clone());
 
         // Randomly select `n_lines` Entry Table lines such that an average number of
         // `num_reindexing_before_full_set` calls to `compact` are needed to compact all
         // the chain table.  See `documentation/Findex.pdf` and coupon problem to
         // understand this formula.
-        let entry_table_length = entry_table.len() as f64;
+        let entry_table_length = all_uids.len() as f64;
         let n_lines = ((entry_table_length * (entry_table_length.log2() + 0.58))
             / f64::from(num_reindexing_before_full_set))
         .ceil() as usize;
-        let entry_table_items_to_reindex = entry_table
-            .keys()
+        let entry_table_uids_to_reindex = all_uids
+            .iter()
             .choose_multiple(&mut rng, n_lines)
             .into_iter()
             .cloned()
-            .collect::<Vec<_>>();
+            .collect::<HashSet<_>>();
 
-        // Unchain the Entry Table entries to be reindexed.
-        let kwi_chain_table_uids = entry_table
-            .unchain::<BLOCK_LENGTH, KMAC_KEY_LENGTH, DEM_KEY_LENGTH, KmacKey, DemScheme>(
-                entry_table_items_to_reindex.iter(),
-                usize::MAX,
-            );
+        println!("to_reindex: {entry_table_uids_to_reindex:?}");
 
-        //
-        // Batch fetch chains from the Chain Table. It's better for performances and
-        // prevents the database. There is no way for the database to know which Entry
-        // Table lines are linked to the requested UIDs since the Entry Table was fetch
-        // entirely and a random portion of it is being compacted.
-        //
-        let chains_to_reindex = self.batch_fetch_chains(&kwi_chain_table_uids).await?;
+        for i in (0..all_uids_vec.len()).step_by(fetch_entry_batch_size) {
+            let slice_end = min(i + fetch_entry_batch_size, all_uids_vec.len());
+            let slice_uids: HashSet<Uid<UID_LENGTH>> =
+                all_uids_vec[i..slice_end].iter().cloned().collect();
+            println!("batch loop: {i}, slice: {slice_uids:?}");
 
-        //
-        // Remove all reindexed Chain Table items. Chains are recomputed entirely.
-        //
-        let chain_table_uids_to_remove = chains_to_reindex
-            .values()
-            .flat_map(|chain| chain.iter().map(|(k, _)| k))
-            .cloned()
-            .collect::<HashSet<_>>();
+            let encrypted_entry_table = self.fetch_entry_table(&slice_uids).await?;
+            println!("table: {encrypted_entry_table:?}");
```

Comment:
Could `compact()` look like that?
```
let all_uids = fetch_entry_table_uids();
for uid_batch in all_uids.by_batch() {
    let entry_table_batch = fetch_entry_table(uid_batch).await?;
    compact_subroutine(entry_table_batch)
}
```

*Note*: this is pseudo-code, it seems there is no function such as `.by_batch()` in the stable version, or at least I couldn't find one.